# Construction of pYPK0_PDC1_ScTKL1_RPS19b (pMEC1131)

[pYPKa_Z_PDC1](pYPKa_Z_PDC1.ipynb)

[pYPKa_E_RPS19b](pYPKa_E_RPS19b.ipynb) 

In [1]:
from pydna.all import *
from pydna import ipynb_importer

In [2]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [3]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [4]:
pYPK0 =read("pYPK0.gb")

In [5]:
promoter_clone = pYPKa_Z_PDC1 =read("pYPKa_Z_PDC1.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nucleotide/NM_001184171.1) record

In [6]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/9169875 width=100% height=250></iframe>')

In [7]:
gb =Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("NM_001184171.1")

In [8]:
gene_template.list_features()

+----------+-----------+-------+-------+--------+--------------+--------+------+
| Feature# | Direction | Start |  End  | Length | id           | type   | orf? |
+----------+-----------+-------+-------+--------+--------------+--------+------+
| 0        |    -->    |   0   |  2043 |   2043 | <unknown id> | source | yes  |
| 1        |    -->    |   <0  | >2043 |   2043 | <unknown id> | gene   | yes  |
| 2        |    -->    |   0   |  2043 |   2043 | <unknown id> | CDS    | yes  |
+----------+-----------+-------+-------+--------+--------------+--------+------+

In [9]:
orf = gene_template.extract_feature(2)

In [10]:
orf.isorf()

True

In [11]:
print(len(orf))
orf.seguid()

2043


m6HK97EKOlSwRjnm3ayQsKUgsDs

In [12]:
print(str(orf.seq.translate()))

MTQFTDIDKLAVSTIRILAVDTVSKANSGHPGAPLGMAPAAHVLWSQMRMNPTNPDWINRDRFVLSNGHAVALLYSMLHLTGYDLSIEDLKQFRQLGSRTPGHPEFELPGVEVTTGPLGQGISNAVGMAMAQANLAATYNKPGFTLSDNYTYVFLGDGCLQEGISSEASSLAGHLKLGNLIAIYDDNKITIDGATSISFDEDVAKRYEAYGWEVLYVENGNEDLAGIAKAIAQAKLSKDKPTLIKMTTTIGYGSLHAGSHSVHGAPLKADDVKQLKSKFGFNPDKSFVVPQEVYDHYQKTILKPGVEANNKWNKLFSEYQKKFPELGAELARRLSGQLPANWESKLPTYTAKDSAVATRKLSETVLEDVYNQLPELIGGSADLTPSNLTRWKEALDFQPPSSGSGNYSGRYIRYGIREHAMGAIMNGISAFGANYKPYGGTFLNFVSYAAGAVRLSALSGHPVIWVATHDSIGVGEDGPTHQPIETLAHFRSLPNIQVWRPADGNEVSAAYKNSLESKHTPSIIALSRQNLPQLEGSSIESASKGGYVLQDVANPDIILVATGSEVSLSVEAAKTLAAKNIKARVVSLPDFFTFDKQPLEYRLSVLPDNVPIMSVEVLATTCWGKYAHQSFGIDRFGASGKAPEVFKFFGFTPEGVAERAQKTIAFYKGDKLISPLKKAF*


In [13]:
pf, pr =parse('''>TKL_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGACTCAATTCACTGACATTGATAAG
                        >TKL_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAGAAAGCTTTTTTCAAAGGAGAAATTA''')

In [14]:
g =pcr( pf, pr, orf)

In [15]:
g.figure()

                                        5ATGACTCAATTCACTGACATTGATAAG...TAATTTCTCCTTTGAAAAAAGCTTTCTAA3
                                                                       ||||||||||||||||||||||||||||| tm 55.6 (dbd) 65.0
                                                                      3ATTAAAGAGGAAACTTTTTTCGAAAGATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGACTCAATTCACTGACATTGATAAG3
                                         ||||||||||||||||||||||||||| tm 56.7 (dbd) 64.4
                                        3TACTGAGTTAAGTGACTGTAACTATTC...ATTAAAGAGGAAACTTTTTTCGAAAGATT5

In [16]:
terminator_clone = pYPKa_E_RPS19b =read("pYPKa_E_RPS19b.gb")

In [17]:
p  =pcr( p167, p567, promoter_clone)
t  =pcr( p568, p166, terminator_clone)

In [18]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [19]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Amplicon(1117), Amplicon(2121), Amplicon(731))

In [20]:
asm =Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [21]:
asm

Assembly:
Sequences........................: [5681] [1117] [2121] [731]
Sequences with shared homologies.: [5681] [1117] [731] [2121]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [9404]
Linear products..................: [9502] [9465] [9454] [9441] [8771] [8435] [7370] [6700] [6351] [3882] [3188] [2815] [98] [61] [50] [37]

In [22]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|1117bp_PCR_prod|38
|                           \/
|                           /\
|                           38|2121bp_PCR_prod|35
|                                              \/
|                                              /\
|                                              35|731bp_PCR_prod|61
|                                                                \/
|                                                                /\
|                                                                61-
|                                                                   |
 -------------------------------------------------------------------

In [23]:
result = candidate.synced(pYPK0)

The new construct should have cseguid ```m-7lOIJ60aZl4b7rN4vanlD7OWk``` and 9404 bp.

In [24]:
print(len(result))
result.cseguid()

9404


m-7lOIJ60aZl4b7rN4vanlD7OWk

In [25]:
result.name = "pMEC1131"
result.description = "pYPK0_PDC1_ScTKL1_RPS19btp"
result.stamp()

cSEGUID_m-7lOIJ60aZl4b7rN4vanlD7OWk

In [26]:
result.write("pYPK0_PDC1_ScTKL1_RPS19b.gb")

pYPK0_PDC1_ScTKL1_RPS19b.gb

###[Download](pYP0 ... .gb) genbank file.

In [27]:
from pydna.all import *

In [28]:
reloaded =read("pYPK0_PDC1_ScTKL1_RPS19b.gb")

In [29]:
reloaded.cseguid()

m-7lOIJ60aZl4b7rN4vanlD7OWk

In [30]:
reloaded.description

'pYPK0_PDC1_ScTKL1_RPS19btp cSEGUID_m-7lOIJ60aZl4b7rN4vanlD7OWk_2015-07-15T12:36:04.081037'